projects/core/directives/hint/hint-position.directive.ts

```diff
@@ -112,7 +112,9 @@ export class TuiHintPosition extends TuiPositionAccessor {
             top > GAP &&
             left > GAP &&
             top + height < viewport.bottom - GAP &&
-            left + width < viewport.right - GAP
+            left + width < viewport.right - GAP &&
+            left > viewport.left + GAP &&
```

Comment:
You can just add viewport to lines 112 && 113

Reply:
fixed